src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Serialization/Object.WriteTests.cs

```diff
@@ -146,5 +148,79 @@ public static void WriteObjectWithNumberHandling()
             var options = new JsonSerializerOptions { NumberHandling = JsonNumberHandling.AllowReadingFromString };
             JsonSerializer.Serialize(new object(), options);
         }
+
+        /// <summary>
+        /// This test is constrained to run on Windows and MacOSX because it causes
+        /// problems on Linux due to the way deferred memory allocation works. On Linux, the allocation can
+        /// succeed even if there is not enough memory but then the test may get killed by the OOM killer at the
```

Comment:
does it get killed inside of JSON or during construction of argument passed in to JSON? can same happen during deserialization or just serialization?

Reply:
It would get killed within the serializer. I don't think we have a test yet but I imagine it could happen on deserialization as well depending on the input size and target type. The comments are borrowed from a writer test that does similar writing of a large payload - https://github.com/dotnet/runtime/blob/5a2d0db89ddfbfe06caa7da314a4fd527954f565/src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Utf8JsonWriterTests.cs#L737-L743.

Reply:
We discuss memory concerns on deserialization more here: https://github.com/dotnet/runtime/blob/main/src/libraries/System.Text.Json/docs/ThreatModel.md#deserialization.